mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/scheduler/ResourceClusterAwareSchedulerActor.java

```diff
@@ -120,6 +135,58 @@ public Receive createReceive() {
             .build();
     }
 
+    private void onBatchScheduleRequestEvent(BatchScheduleRequestEvent event) {
+        log.info("Received batch schedule request event: {}", event);
+
+        // If the size of the batch request is 1 we'll handle it as the "old" schedule request
+        if (event.getRequest().getScheduleRequests().size() == 1) {
+            final ScheduleRequestEvent scheduleRequestEvent = ScheduleRequestEvent.of(event);
+            self().tell(scheduleRequestEvent, self());
+        } else {
+            if (event.isRetry()) {
+                log.info("Retrying Batch Schedule Request {}, attempt {}", event.getRequest(),
+                    event.getAttempt());
+            }
+
+            CompletableFuture<Object> assignedFuture =
+                resourceCluster
+                    .getTaskExecutorsFor(event.getTaskExecutorAllocationRequests())
+                    .<Object>thenApply(event::onAssignment)
+                    .exceptionally(event::onFailure);
+
+            pipe(assignedFuture, getContext().getDispatcher()).to(self());
+        }
+    }
+
+    private void onAssignedBatchScheduleRequestEvent(AssignedBatchScheduleRequestEvent event) {
+        // Each and every allocation of a batch schedule request will be handled separately (ie. assignment, retry etc etc)
+        event
+                .getAllocations()
+                .forEach((key, value) -> {
+                    final ScheduleRequest scheduleRequest = event.getScheduleRequestEvent().getAllocationRequestScheduleRequestMap().get(key);
+                    final ScheduleRequestEvent scheduleRequestEvent = ScheduleRequestEvent.of(scheduleRequest, event.getScheduleRequestEvent().eventTime);
+                    final AssignedScheduleRequestEvent assignedScheduleRequestEvent = new AssignedScheduleRequestEvent(scheduleRequestEvent, value);
+
+                    self().tell(assignedScheduleRequestEvent, self());
+                });
+    }
+
+    private void onFailedToBatchScheduleRequestEvent(FailedToBatchScheduleRequestEvent event) {
+        batchSchedulingFailures.increment();
+        if (event.getAttempt() >= this.maxScheduleRetries) {
+            log.error("Failed to submit the batch request {} because of ", event.getScheduleRequestEvent(), event.getThrowable());
```

Comment:
in this case does it need to route some message/status back to the job actor?

Reply:
I'm following the behavior we currently have during scheduling ops -> https://github.com/Netflix/mantis/blob/81bce548e92f5f4cc048f07a23acc76a26941717/mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/scheduler/ResourceClusterAwareSchedulerActor.java#L206

Reply:
do you think we should change this behavior? If so, how?

Reply:
i guess in this case there is no more retry/actions to this job so maybe we should mark the job as failed instead? So users won't feel that the job is still being handled.